### ISSUE TITLE
AdGuard redesign tracking issue

### ISSUE THREAD

**Issue by ameshkov** [AdguardTeam/AdGuardHome#2554]
It may be necessary to consult this issue: https://github.com/AdguardTeam/AdGuardHome/issues/2075

This issue just contains a checklist of pages that we implement (see "client2").

* [X] Initial setup
* [ ] Login
* [ ] Dashboard
* [ ] Setup guide - part 1 (before review)
* [ ] Setup guide - part 2 (after review)
* [ ] DNS blocklists
* [ ] DNS whitelists
* [ ] DNS rewrites
* [ ] Blocked services
* [ ] Custom filtering rules
* [ ] Settings -> General settings
* [ ] Settings -> DNS settings
* [ ] Settings -> Encryption
* [ ] Settings -> Client settings
* [ ] Settings -> DHCP
* [ ] Query log
* [ ] Global notifications
    * [ ] New version is available
    * [ ] Certificate is about to expire

**Comment by Blakhard** [comment]
For now, we can not make date picker on Dashboard due to API.

**Comment by ameshkov** [comment]
@Blakhard yep, np, this feature will be moved to a future version.

**Comment by Blakhard** [comment]
1) Now "Blocked ads" and "Blocked trackers" are combined to "Blocked by filters". 
2) No chart data for "Average processing time"
3) "Internal server statistic" block can not be done, appear in next API releases.
4) "Server uptime" in header, unclear is it server uptime, or machine uptime.

**Comment by ameshkov** [comment]
@Blakhard we'll have to implement these in this version so some changes to the backend.

**Comment by WildByDesign** [comment]
@ameshkov Does this redesign implementation allow the potential for adding Dark Mode to UI elements?

Since you are at the beginning of the implementation and design stage for this, I figured I should bring this up. Dark Mode continues to be one of the most up-voted feature requests.

Thank you for all of the great work that your team does. Despite the whole Covid-19 slowing down much of the world, your team kept on rocking it without slowing down. Cheers!

**Comment by ameshkov** [comment]
Yep, I guess once the redesign is completed, we'll be able to bring the dark mode soon after that

**Comment by Aikatsui** [comment]
@ameshkov Is it possible to try https://github.com/AdguardTeam/AdGuardHome/issues/481#issuecomment-546758541 ?
(I know, already has many issues to address but very difficult to work with query log to view time series data)

**Comment by ameshkov** [comment]
Well, I'd better do that later, we already planned too much on this redesign.

**Comment by ainar-g** [comment]
#3066 proposes adding autoupdate of the query log as well.

**Comment by gregtwallace** [comment]
For the dashboard and charts, have you guys considered a more broad and user controllable scheme for the graphs?  That is, I was looking at refactoring the code so any rule (whether DNS blacklist, blocked service, or custom rule) would have a user selectable option for the type of content being blocked (e.g. based on the current graphs: Ads, Malicious, and Content) and then possibly even a sub-type (e.g. Ads ->Trackers, Malicious -> Abuse, Malicious -> AdGuard Security Service, Content -> Pornography, Content -> AdGuard Parental Control) and maybe even make the sub-type a user editable field rather than drop down.

Basically this would make the reporting customizable so the user can define and then see what type of content is being blocked and drill further into a graph to look at the sub-types of content.  This would be advanced functionality but existing filter lists could just be crafted to have a pre-defined type/sub-type so "regular" users don't have to do any extra work compared to the current setup.

I was contemplating a refactor to make this work but didn't want to dive down that hole if you guys already planned something like this or were going a different direction.

**Comment by ncodee** [comment]
Sorry to ask, but it's been a while, and wanted to ask if the development had come to an hault, or it just hasn't been updated here?

**Comment by Gontier-Julien** [comment]
Hi,
Alright first I know there is already a lot of questions around the dark mode, so I've decided to take care of it if needed.

I don't know if you have taken care of it in your private repo, I know it not the top priority to do the redesign and the dark mode, but I would be happy to get my hands dirty to get it done ^^
